app/src/main/java/com/example/taskyy/data/remote/TaskyyApi.kt

```diff
@@ -4,19 +4,25 @@ import com.example.taskyy.data.local.room_entity.EventEntity
 import com.example.taskyy.data.local.room_entity.ReminderEntity
 import com.example.taskyy.data.local.room_entity.TaskEntity
 import retrofit2.http.DELETE
+import retrofit2.http.Field
+import retrofit2.http.FormUrlEncoded
 import retrofit2.http.GET
 import retrofit2.http.Multipart
 import retrofit2.http.POST
 import retrofit2.http.PUT
 import retrofit2.http.Query
 
 interface TaskyyApi {
-
+    @FormUrlEncoded
     @POST("/register")
-    fun registerUser(@Query("key") key: String)
+    suspend fun registerUser(
+        @Field("fullName") fullName: String,
+        @Field("email") email: String,
+        @Field("password") password: String
```

Comment:
The /register endpoint expects the request as a JSON body. You're attaching the strings as separate fields here. To fix that you can do this:
```
// In auth->data->request for example
data class RegisterRequest(
    val fullName: String,
    val email: String,
    val password: String
)
```
And then your Retrofit function:
```
@POST("/register")
suspend fun registerUser(@Body body: RegisterRequest)
```